Sources/Interpreter/BuiltinValue.swift

```diff
@@ -76,7 +76,7 @@ enum BuiltinValue {
 /// Methods to create builtin value from IR constants.
 extension BuiltinValue {
 
-  /// Creates instance of builtin value with integer constant `c`.
+  /// Creates instance of builtin value with `c`.
   public init(withIntegerConstant c: IntegerConstant) {
```

Comment:
`withIntegerConstant` is needlessly putting type information in the function name.  Is this a value-preserving conversion?  In that case:


```suggestion
  public init(_ c: IntegerConstant) {
```

I'm not sure the name `IntegerConstant` is good either, but that's a separate issue.

Reply:
Assuming value-preserving means preserving mathematical value, then yes its a value preserving conversion.